godot-macros/src/util/list_parser.rs

```diff
@@ -198,6 +198,33 @@ impl ListParser {
         }
     }
 
+    pub(crate) fn next_allowed_key_optional_value(
+        &mut self,
+        allowed_bool_keys: &[&str],
+        allowed_kv_keys: &[&str],
+    ) -> ParseResult<Option<(Ident, Option<KvValue>)>> {
```

Comment:
Would be good to keep terminology consistent everywhere:
- "flag/flags" for standalone boolean keys
- "kv/kvs" for key-value pairs

As for the method name, the "optional" is a bit unclear (I know it refers to "value", but had to read multiple times). Maybe it's not needed given the return type has already `Option`? So, `next_allowed_flag_or_kv` might be a different way to state it... or `next_allowed_key_value`.

Is "allowed" needed at all?

Reply:
I'll use flags, sure.

So the though behind the name was that it's doing the same thing as `next_key_optional_value`, but with the possible flag names and kv keys specified, similar to `next_allowed_ident(allowed_ids)`.  So combining these two names, I landed on what's here.  I agree that the name is a lot to parse, but I think it's reasonable to a user of this struct.  An API user here mostly likely will be learning the API by reading the source and will first discover those methods I'm basing this name on before this one.  It might be reasonable to rethink the naming, but I don't think that's in scope for this PR.

The outer `Option`  is `Some` when there is a value to parse and `None` after then end of the sequence, similar to `std::iter::Iterator::next()`.  If a not-allowed ident is a key, it'll return `Err`, so the `Option` type doesn't come into play.  This is consistent with `next_allowed_ident`.

So you're right that this is a bit funny, but it's because I'm combining a couple of existing funny things; I don't think I added any funniness.  I will add a docstring to clarify the function, though.